pkg/varlinkapi/attach.go

```diff
@@ -45,22 +45,24 @@ func (i *LibpodAPI) Attach(call iopodman.VarlinkCall, name string, detachKeys st
 	var finalErr error
 	resize := make(chan remotecommand.TerminalSize)
 	errChan := make(chan error)
+	varlink := VarlinkCall{&call}
 
-	if !call.WantsUpgrade() {
-		return call.ReplyErrorOccurred("client must use upgraded connection to attach")
+	if ok, err := varlink.RequiresUpgrade(); !ok {
+		return varlink.ReplyErrorOccurred(err.Error())
```

Comment:
wwyt about defining an actual error called RequiresUpgrade or some such for a return here?

Reply:
@baude That was my fall back.  I'll put that together now.